packages/core/src/components/button/test/basic/button.e2e.ts

```diff
@@ -3,97 +3,40 @@ import { expect } from '@playwright/test';
-    const buttonBackgroundColor = await innerButton.evaluate(
-      (button2) => getComputedStyle(button2).backgroundColor,
-    );
-    expect(buttonBackgroundColor).toBe('rgb(15, 50, 99)');
-  });
-
-  test('component should have correct border when active', async ({ page }) => {
-    await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    const innerButton = button.locator('button');
-    await innerButton.focus();
-    const buttonBackgroundColor = await innerButton.evaluate(
-      (button2) => getComputedStyle(button2).backgroundColor,
-    );
-    const buttonBorderColor = await innerButton.evaluate(
-      (button2) => getComputedStyle(button2).borderColor,
-    );
-    const buttonOutlineColor = await innerButton.evaluate(
-      (button2) => getComputedStyle(button2).outlineColor,
-    );
-    expect(buttonBackgroundColor).toBe('rgb(43, 112, 211)');
-    expect(buttonBorderColor).toBe('rgb(22, 65, 127)');
-    expect(buttonOutlineColor).toBe('rgb(22, 65, 127)');
-  });
-
-  test('have Type = Button', async ({ page }) => {
-    await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    const innerButton = button.locator('button');
-    const type = await innerButton.getAttribute('type');
-    expect(type).toBe('button');
-  });
-
-  test('check if the inner button contain classes primary lg and sc-tds-button', async ({
-    page,
-  }) => {
+  test('expect button to be of role button', async ({ page }) => {
     await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    const innerButton = button.locator('button');
-    const classAttribute = await innerButton.getAttribute('class');
-    const classes = classAttribute.split(' ');
-    expect(classes).toContain('primary');
-    expect(classes).toContain('lg');
-    expect(classes).toContain('sc-tds-button');
+    const button = page.getByRole('button');
+    await expect(button).toHaveCount(1);
   });
 
-  test('Text is displayed, style is aligned to default(primary) state', async ({ page }) => {
+  test('Text is displayed', async ({ page }) => {
     await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    await expect(button).toHaveCSS('color', 'rgb(13, 15, 19)');
-    const textAttribute = await button.textContent();
-    expect(textAttribute).toBe('Button');
+    const button = page.getByText('Button', { exact: true });
+    await expect(button).toBeVisible();
   });
 
   test('Check so that height is correct to lg/default measurements', async ({ page }) => {
     await page.goto(componentTestPath);
-    const button = page.locator('tds-button');
-    const innerButton = button.locator('button');
-    const buttonHeight = await innerButton.evaluate((button2) => getComputedStyle(button2).height);
+    const button = page.getByText('Button', { exact: true });
+    const buttonHeight = await button.evaluate((style) => getComputedStyle(style).height);
```

Comment:
Nice!

Reply:
Yeah not checking the height might be a breaking change in some applications using the code so needs to be checked :D